module/VuFind/src/VuFind/Search/EDS/Options.php

```diff
@@ -148,23 +193,68 @@ public function __construct(
     }
 
     /**
-     * Get an array of search mode options
+     * Basic 'getter' for advanced search handlers.
      *
      * @return array
      */
-    public function getModeOptions()
+    public function getAdvancedHandlers()
     {
-        return $this->modeOptions;
+        return $this->getApiProperty('advancedHandlers');
     }
 
     /**
-     * Get the default search mode
+     * Basic 'getter' for basic search handlers.
+     *
+     * @return array
+     */
+    public function getBasicHandlers()
+    {
+        return $this->getApiProperty('basicHandlers');
+    }
+
+    /**
+     * Get default search handler.
      *
      * @return string
      */
-    public function getDefaultMode()
+    public function getDefaultHandler()
+    {
+        $this->setOptionsFromApi();
+        return parent::getDefaultHandler();
+    }
+
+    /**
+     * Given a basic handler name, return the corresponding label (or false
+     * if none found):
+     *
+     * @param string $handler Handler name to look up.
+     *
+     * @return string
+     */
+    public function getLabelForBasicHandler($handler)
```

Comment:
Should we make this change in the base class to make extensibility cleaner, rather than overriding it here?

Reply:
Yes, moved.